homeassistant/components/zwave_js/config_flow.py

```diff
@@ -481,12 +509,32 @@ async def async_step_configure_addon(
             return await self.async_step_start_addon()
 
         usb_path = self.usb_path or addon_config.get(CONF_ADDON_DEVICE) or ""
-        network_key = addon_config.get(CONF_ADDON_NETWORK_KEY, self.network_key or "")
+        s0_legacy_key = addon_config.get(
+            CONF_ADDON_S0_LEGACY_KEY, self.s0_legacy_key or ""
+        )
+        s2_access_control_key = addon_config.get(
+            CONF_ADDON_S2_ACCESS_CONTROL_KEY, self.s2_access_control_key or ""
+        )
+        s2_authenticated_key = addon_config.get(
+            CONF_ADDON_S2_AUTHENTICATED_KEY, self.s2_authenticated_key or ""
+        )
+        s2_unauthenticated_key = addon_config.get(
+            CONF_ADDON_S2_UNAUTHENTICATED_KEY, self.s2_unauthenticated_key or ""
+        )
 
         data_schema = vol.Schema(
             {
                 vol.Required(CONF_USB_PATH, default=usb_path): str,
-                vol.Optional(CONF_NETWORK_KEY, default=network_key): str,
+                vol.Optional(CONF_S0_LEGACY_KEY, default=s0_legacy_key): str,
```

Comment:
We validate it as string. Should we already validate it as a valid key format? So either empty or 32 chars ?

Reply:
Yeah, but those kinds of changes is for another PR. This is just so that we retain key management in the integration.